samza-api/src/main/java/org/apache/samza/system/StreamSpec.java

```diff
@@ -200,4 +200,8 @@ public String get(String propertyName) {
   public String getOrDefault(String propertyName, String defaultValue) {
     return config.getOrDefault(propertyName, defaultValue);
   }
+
+  public SystemStream getSystemStream() {
```

Comment:
Make sense. So, that probably means that all usage as "unique id" for a stream should be changed to use StreamSpec.getId() in the programming layer, and SystemStream should only be visible by the ExecutionEnvironment. I will make the change.